tasks/task_manager.cc

```diff
@@ -250,6 +250,10 @@ bool task_manager::task::is_complete() const noexcept {
     return _impl->is_complete();
 }
 
+void task_manager::task::release_resources() noexcept {
```

Comment:
Ah, so before only compaction tasks had this method, but now all tasks can have it? That is why `compaction/` is also changed?

Reply:
Agree, but now that this is a generic thing across tasks, can we invoke it automatically after the task finishes, and before it is entering the TTL phase? I see below this is done manually. Can be a follow-up if this is not simple, no need to delay fixing this bug.